mysql/resource_rds_config.go

```diff
@@ -0,0 +1,182 @@
+package mysql
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"log"
+	"strconv"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+
+	"github.com/go-sql-driver/mysql"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceRDSConfig() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: CreateRDSConfig,
+		UpdateContext: UpdateRDSConfig,
+		ReadContext:   ReadRDSConfig,
+		DeleteContext: DeleteRDSConfig,
+		Importer: &schema.ResourceImporter{
+			StateContext: ImportRDSConfig,
+		},
+		Schema: map[string]*schema.Schema{
+			"binlog_retention_period": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Default:     0,
+				Description: "Sets the number of hours to retain binary log files",
+			},
+			"replication_target_delay": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Default:     0,
+				Description: "Sets the number of seconds to delay replication from source database instance to the read replica",
+			},
+		},
+	}
+}
+
+func CreateRDSConfig(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	db, err := getDatabaseFromMeta(ctx, meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	for _, stmtSQL := range RDSConfigSQL(d) {
+		log.Println("Executing statement:", stmtSQL)
+
+		_, err = db.ExecContext(ctx, stmtSQL)
+		if err != nil {
+			return diag.Errorf("failed running SQL to set RDS Config: %v", err)
+		}
+	}
+
+	id := strconv.FormatInt(time.Now().UTC().UnixNano(), 10)
+
+	d.SetId(id)
+
+	return nil
+}
+
+func UpdateRDSConfig(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	db, err := getDatabaseFromMeta(ctx, meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	for _, stmtSQL := range RDSConfigSQL(d) {
+		log.Println("Executing statement:", stmtSQL)
+
+		_, err = db.ExecContext(ctx, stmtSQL)
+		if err != nil {
+			return diag.Errorf("failed updating RDS config: %v", err)
+		}
+	}
+
+	return nil
+}
+
+func ReadRDSConfig(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	db, err := getDatabaseFromMeta(ctx, meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	stmtSQL := "call mysql.rds_show_configuration"
+
+	log.Println("Executing query:", stmtSQL)
+	rows, err := db.QueryContext(ctx, stmtSQL)
+	if err != nil {
+		if mysqlErr, ok := err.(*mysql.MySQLError); ok {
+			if mysqlErr.Number == unknownDatabaseErrCode {
```

Comment:
Why would unknown DB mean this resource doesn't exist?

Or when does that error happen?

Reply:
You are right, not needed. Copy/pasted from another block.